apps/meteor/client/views/room/hooks/useUserInfoActions/useUserInfoActions.ts

```diff
@@ -40,6 +42,14 @@ export const useUserInfoActions = (
 	reload?: () => void,
 	size = 2,
 ): { actions: [string, UserInfoAction][]; menuActions: any | undefined } => {
+	const { data, refetch, isSuccess: membershipCheckSuccess } = useMemberExists(rid, user.username as string);
```

Comment:
This `useUserInfoActions` hook can be called from 2 places - 

- Room Members Contextual Bar
- User Card on Mentions and stuff.

If it is being called from Room Members Contextual Bar, it will be called for each and every user in the list. It will inturn call the `rooms.isMember` endpoint for each and every user in the list, while we already know that if the user is present in room members contextual bar, user is already part of the room. In this case we are unnecessarily calling the `rooms.isMember` endpoint for each user.

This flow only makes sense when we call this hook from UserCard or some other place, where we are not user about user's subscription to the room.

I would suggest to move up this check to outside this hook, to it's parent and this hook should receive this info as a prop, whether user is part of room or not, based on this prop, we can return the list of user actions.

Reply:
Hey @yash-rajpal  Thanks for noticing this. I agree we should not call the endpoint when we already know if the user is a member.
To fix this I have added an override prop for the useMemberExists hook which will prevent api call when an override value is provided. And when calling this hook in memberlist, i am passing the override value and hence preventing unnecessary api calls.